lib/canonical_version.sh

```diff
@@ -1,4 +1,28 @@
-canonicalVersion() {
+#!/usr/bin/env bash
+
+fetch_erlang_versions() {
+  url="https://raw.githubusercontent.com/HashNuke/heroku-buildpack-elixir-otp-builds/master/otp-versions"
+  curl -s "$url" 
+}
+
+exact_erlang_version_available() {
+  version=$1
+  available_versions=$2
+  found=1
+  while read -r line; do
+    if [ "$line" = "$version" ]; then
+      found=0
+    fi
+  done <<< "$available_versions"
+  echo $found
+}
+
+check_erlang_version() {
   version=$1
-  echo $version
+  exists=$(exact_erlang_version_available "$version" "$(fetch_erlang_versions)")
+  if [ $exists -ne 0 ]; then
+    output_line "Sorry, Erlang $version isn't supported yet. For a list of supported versions, please see https://github.com/HashNuke/heroku-buildpack-elixir#version-support"
+    exit 1
+  fi
```

Comment:
after this `if` statement is where `echo $version` should be

Reply:
Sorry, I don't totally follow. Why do we need to echo the version at all?

Reply:
just because it was there before and you're echoing either `1` or `0` now instead which makes even less sense to echo out

Reply:
Oh, sorry, I'm not seeing it in this pull request's diff anywhere. Do you mean the version that gets output here?
https://github.com/HashNuke/heroku-buildpack-elixir/blob/79b8b58114c421d22f0cf4f85c79e2b01098a64c/lib/misc_funcs.sh#L42

Reply:
No I'm talking about the `echo $version` shown in red above

Reply:
Oh I see what you're referring to. I think that was some code in an intermediate commit within the pull request. This file doesn't exist on master so the diff for this PR does not have the echo, but one of the earlier commits added an echo and another commit deleted it. Should be safe to ignore.